internal/controller/testworkflows/testworkflow_controller.go

```diff
@@ -46,9 +53,162 @@ type TestWorkflowReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *TestWorkflowReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// Delete CronJobs if it were created for deleted Test Workflow
+	var testWorkflow testworkflowsv1.TestWorkflow
+	if err := r.Get(ctx, req.NamespacedName, &testWorkflow); err != nil {
+		if errors.IsNotFound(err) {
+			if err = r.CronJobClient.DeleteAll(ctx,
+				cronjob.GetSelector(req.NamespacedName.Name, cronjob.TestWorkflowResourceURI), req.NamespacedName.Namespace); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	events := testWorkflow.Spec.Events
+	for _, template := range testWorkflow.Spec.Use {
+		var testWorkflowTemplate testworkflowsv1.TestWorkflowTemplate
+		if err := r.Get(ctx, types.NamespacedName{Namespace: testWorkflow.Namespace, Name: template.Name}, &testWorkflowTemplate); err != nil {
+			if errors.IsNotFound(err) {
+				continue
+			}
+
+			return ctrl.Result{}, err
+		}
+
+		events = append(events, testWorkflowTemplate.Spec.Events...)
+	}
+
+	hasTemplates := len(testWorkflow.Spec.Use) != 0
+	_, ok := testWorkflow.Labels[cronjob.TestWorkflowTemplateResourceURI]
+	if ok && !hasTemplates {
+		delete(testWorkflow.Labels, cronjob.TestWorkflowTemplateResourceURI)
+		return ctrl.Result{}, r.Update(ctx, &testWorkflow)
+	}
+
+	if !ok && hasTemplates {
+		if testWorkflow.Labels == nil {
+			testWorkflow.Labels = make(map[string]string)
+		}
+
+		testWorkflow.Labels[cronjob.TestWorkflowTemplateResourceURI] = "yes"
+		return ctrl.Result{}, r.Update(ctx, &testWorkflow)
+	}
+
+	newCronJobConfigs := make(map[string]*testworkflowsv1.CronJobConfig)
```

Comment:
Not sure, but I think that it should be rather a list - you may want to have multiple cronjobs with the same schedule, but different config parameters

Reply:
If we add config parameters, then we will need to switch to list

Reply:
yes, like, we will delete the old cron job without config and create a few new ones with configs, or might be, keep old one if we have cron jobs with emoty configs and add new ones with configs